bears/vcs/git/GitCommitBear.py

```diff
@@ -123,6 +127,10 @@ def run(self, allow_empty_commit_message: bool = False, **kwargs):
                 yield Result(self, 'HEAD commit has no message.')
             return
 
+        if not allow_revert_commits and shortlog.startswith('Revert'):
```

Comment:
this isnt sufficient.  it is a good heuristic to start the checking process, but it needs to be more detailed.  See https://github.com/coala/coala-bears/issues/337#issuecomment-372045150

Reply:
Ping @kriti21 , please fix this first.  Until you're correctly verifying that it is a revert commit, everything else in the PR is not worth reviewing.

Reply:
According to https://github.com/coala/coala-bears/issues/337#issuecomment-372045150 , the shortlog must be of the format `Revert "<reverted shortlog>"` implying that it must begin with `Revert` and the following string could be anything. This is the basis of line #136 above. 

In the line #139 below, the expected first line of commit body i.e. `This reverts commit <sha>.` is separated and the rest should be reason. 

I am not getting why and how to change this line (#136) above. If a check is applied to verify that the headline starts with `This reverts commit ` and contains a valid commit sha, this should work fine IMO. Could you please correct me where I am getting this wrong? @jayvdb

Reply:
Because `shortlog.startswith('Revert')` is insufficient for the shortlog.  Please precisely check the shortlog.